src/runtime_config/constitution.js

```diff
@@ -5,64 +5,85 @@ class Action {
   }
 }
 
+function checkType(value, type, field) {
+  const optional = type.endsWith("?");
+  if (optional) {
+    if (typeof value === null || typeof value === undefined) {
+      return;
+    }
+    type = type.slice(0, -1);
+  }
+  if (
+    (type === "array" && !Array.isArray(value)) ||
+    (type === "integer" && !Number.isInteger(value)) ||
+    typeof value !== type
+  ) {
+    throw new Error(`${field} must be of type ${type} but is ${typeof value}`);
+  }
+}
+
+function checkEnum(value, members, field) {
+  if (!members.contains(value)) {
+    throw new Error(`${field} must be one of ${members}`);
+  }
+}
+
+function checkBounds(value, low, high, field) {
```

Comment:
That always puts NaN in bounds, which isn't wrong as such, but seems counter intuitive to me.

Reply:
Ah, true!